unit-tests/src/test/scala/scala/scalanative/unsafe/ExternSuite.scala

```diff
@@ -50,4 +50,13 @@ object ExternSuite extends tests.Suite {
     Ext2.p(buf2, bufsize, c"%d", 1)
     assert(string.strcmp(buf2, c"1") == 0)
   }
+
+  val cb: CFuncPtr0[CInt] = new CFuncPtr0[Int] {
+    override def apply(): Int = 42
+  }
+  test("allows to use generic function as argument") {
+    val res0 = testlib.exec0(cb) //expected CFuncPtr0[Int]
+    val res1 = testlib.exec(cb)  //expected CFuncPtr
+    assert(res0 == res1)
```

Comment:
Consider making this test more precise by explicitly testing the expected result of each call, rather than testing that the two are equal. Currently, this test would pass even if `exec0` and `exec` both return the same wrong result (e.g., `0`).